README.md

```diff
@@ -1 +1,41 @@
+# Data/translation repository for the ByteLegend programming game
 
+## What is ByteLegend
+
+[ByteLegend](https://bytelegend.com) is a free, open-source MMORPG game where you acquire real-world, high-paying programming skills.
+
+## Welcome
+
+Thank you for your interest in contributing to ByteLegend. This repository is specifically for text data and translations but we assume you're already familiar with the [main readme](https://github.com/ByteLegend/ByteLegend/blob/master/README.md) and the [contributors' guide](https://github.com/ByteLegend/ByteLegend/blob/master/docs/en/CONTRIBUTING.md).  
+
+## How the files work
+
+In this repo you'll spend most of your attention on the [YAML-format](https://en.wikipedia.org/wiki/YAML) `.yml` files which contain the human-entered text, in multiple languages where available.
+As of December 2021, this is mostly English (`EN`) and Simplified Chinese (`ZH_HANS`).
+These manually added strings are then either used verbatim in the game or translated automatically to other languages, as chosen by the player.
+Automatic translation uses Google Translate API, while the [JSON-format](https://en.wikipedia.org/wiki/JSON) `.json` files serve as cache for the translated strings so that the game doesn't make unnecessary API calls.
+
+If you compare an .yml file with a similarly-named .json file, you will find all strings from the .yml file in the .json file but the latter will have more translations.
+These additional translations are generated automatically. This can be either through scripting triggered automatically by your PR being merged or executed manually by developers.
+Either way, you don't have to worry about strings missing from the .json file as long as the structure of the file is valid.
+
+What you have to keep in mind however is that there is currently a possibility that strings in the .yml file are updated but the corresponding cached values are not updated in the .json file.
+As of December 2021, as contributions can still be managed manually, developers do try to catch such instances and regenerate the .json file themselves.
```

Comment:
I just checked [the code](https://github.com/ByteLegend/ByteLegend/blob/5e2d72886e1957e717b1988b0eb3990c16ee8714/utils/src/main/kotlin/com/bytelegend/utils/I18nGenerator.kt#L78) and my memory was wrong previously. In short, people don't need to care `.json` files at all - when you change something in yaml, `json` will be automatically updated:

- If an English text in JSON/YAML doesn't change, the translated texts of other languages in JSON will be used.
- Otherwise, the translated texts in JSON will be updated via Google translation API.


 So I guess we can avoid emphasizing this, just say "only make changes to YAMLs and the automation tool will take cate of everything else".